core/commands/swarm.go

```diff
@@ -476,6 +507,21 @@ func peersWithAddresses(addrs []string) (pis []pstore.PeerInfo, err error) {
 	return pis, nil
 }
 
+func getNode(req cmds.Request, res cmds.Response) *core.IpfsNode {
```

Comment:
I cannot. The 2 errors need different commands.ErrorType set. So I either create my own error type, reuse the commands.Error or have 3 return arguments `(*core.IpfsNode, commands.ErrorType, error)`. Alternatively I inline this method back and wash my hands afterwards.